File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution{
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> num_map = new HashMap<>(); //number, index value
        for (int i = 0; i < nums.length; i++){
            int complement = target - nums[i];
            if (num_map.containsKey(complement)){
                return new int[] {num_map.get(complement),i};
            }
            else{
                num_map.put(nums[i], i);
            }
         }
        return new int[] {};
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        i = 0
        j = len(numbers) - 1
        res = []
        while i < j:
            if numbers[i] + numbers[j] == target:
                res.append(i + 1)
                res.append(j + 1)
                return res
            elif (numbers[i] + numbers[j] < target):
                i += 1
            else:
                j -= 1
        return None

        